libraries/commerce/product/selectors/product.js

```diff
@@ -210,18 +209,23 @@ export const getProductsResult = createSelector(
 );
 
 /**
- * Retrieves the current product name.
+ * Retrieves the product name for the given id.
  * @param {Object} state The current application state.
  * @return {string|null}
  */
 export const getProductName = createSelector(
-  getCurrentProduct,
-  (product) => {
-    if (!product) {
+  getProducts,
+  (state, props) => props.productId,
+  (products, productId) => {
+    if (!productId) {
+      return null;
+    }
+
+    if (!products[productId] || !products[productId].productData) {
```

Comment:
Instead if doing this check all the time, you could cache `products[productId]` and `products[productId].productData` by again utilizing or creating a `getProduct` selector and use that in each of these selectors. And then only check `if (!product)` ...

Reply:
Changed, created a new selector to gather productData